src/main/java/hudson/scm/SubversionRepositoryStatus.java

```diff
@@ -161,10 +174,70 @@ void setJobProvider(JobProvider jobProvider) {
+            SubversionRepoUUIDAndRootPath uuidAndRootPath = null;
+            for (Map.Entry<String, UUID> e : remoteUUIDCache.entrySet()) {
+                String remoteRepoRootURL = e.getKey();
+                String remoteRepoRootURLWithSlash = remoteRepoRootURL + "/";
+                if (urlFromConfiguration.startsWith(remoteRepoRootURLWithSlash) || urlFromConfiguration.equals(remoteRepoRootURL) ) {
+                    UUID uuid = e.getValue();
+                    String rootPath = SVNURL.parseURIDecoded(e.getKey()).getPath();
+                    uuidAndRootPath = new SubversionRepoUUIDAndRootPath(uuid, rootPath);
+
+                    LOGGER.finer("Using cached uuid for module location " + urlFromConfiguration + " of job "+ job);
+                    break;
+                }
+            }
+
+            if (uuidAndRootPath == null) {
+                if (LOGGER.isLoggable(FINER)) {
+                    LOGGER.finer("Could not find " + urlFromConfiguration + " in " + remoteUUIDCache.keySet());
+                }
+                UUID remoteUUID = moduleLocation.getUUID(job, scm);
+                SVNURL repositoryRoot = moduleLocation.getRepositoryRoot(job, scm);
+                remoteUUIDCache.put(repositoryRoot.toString(), remoteUUID);
+                uuidAndRootPath = new SubversionRepoUUIDAndRootPath(remoteUUID, repositoryRoot.getPath());
+            }
+            return uuidAndRootPath;
+        }
+
+        boolean doModuleLocationHasAPathFromAffectedPath(String configuredRepoFullPath, String rootRepoPath, Set<String> affectedPath) {
+            boolean containsAnAffectedPath = false;
+            if( configuredRepoFullPath.startsWith(rootRepoPath) ) {
+                String remainingRepoPath = configuredRepoFullPath.substring(rootRepoPath.length());
+                if(remainingRepoPath.startsWith("/"))   remainingRepoPath=remainingRepoPath.substring(1);
+                String remainingRepoPathSlash = remainingRepoPath + '/';
+
+                for (String path : affectedPath) {
+                    if(path.equals(remainingRepoPath) /*for files*/ || 
+                            path.startsWith(remainingRepoPathSlash) /*for dirs*/ ||
+                            remainingRepoPath.length()==0 /*when someone is checking out the whole repo (that is, configuredRepoFullPath==rootRepoPath)*/) {
+                        // this project is possibly changed. poll now.
+                        // if any of the data we used was bogus, the trigger will not detect a change
+                        containsAnAffectedPath = true;
+                        break;
+                    }
+                }
+            }
+            return containsAnAffectedPath;
+        }
+
+        private void scheduleImediatePollingOfJob(Job job, SCMTrigger trigger, List<SvnInfo> infos) {
+            LOGGER.fine("Scheduling the immediate polling of "+job);
```

Comment:
@kenji21 Please, use `LOGGER.fine("Scheduling the immediate polling of " + job);`